plugins/tuxec/fu-tuxec-internal-device.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2021 Daniel Campello <campello@chromium.org>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-tuxec-device.h"
+#include "fu-tuxec-internal-device.h"
+
+#include <libflashrom.h>
+
+struct _FuTuxecInternalDevice {
+	FuTuxecDevice		 parent_instance;
+};
+
+G_DEFINE_TYPE (FuTuxecInternalDevice, fu_tuxec_internal_device,
+	       FU_TYPE_TUXEC_DEVICE)
+
+static void
+fu_tuxec_internal_device_init (FuTuxecInternalDevice *self)
+{
+	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_INTERNAL);
+	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_UPDATABLE);
+	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_NEEDS_REBOOT);
+	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_REQUIRE_AC);
+	fu_device_add_instance_id (FU_DEVICE (self), "main-system-firmware");
```

Comment:
I think this is usually reserved for BIOS, we're going to end up with two instance ID's of main-system-firmware possibly no?